cmd/agent/app/reporter/grpc/reporter.go

```diff
@@ -86,12 +89,39 @@ func addProcessTags(spans []*model.Span, process *model.Process, agentTags []mod
 	}
 	for _, span := range spans {
 		if span.Process != nil {
-			span.Process.Tags = append(span.Process.Tags, agentTags...)
+			if duplicateTagsPolicy != reporter.Duplicate {
+				for _, agentTag := range agentTags {
+					index, alreadyPresent := checkIfPresentAlready(span.Process.Tags, agentTag)
+					if alreadyPresent {
+						// If Policy is Agent, remove and add else do nothing as client is already present
```

Comment:
nit: `// If Policy is to keep agent tags, purge duplicate client tags. Else discard agent tags.`